source/connect.txt

```diff
@@ -17,8 +28,199 @@ Connect to MongoDB
+
+Specify a Certificate Revocation List (CRL)
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. include:: /includes/connect/crl-tabs.rst
+
+To learn more about specifying a CRL, see :ref:`pymongo-crl` in
+the TLS configuration guide.
+
+Present a Client Certificate
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. include:: /includes/connect/client-cert-tabs.rst
+
+To learn more about specifying a client certificate, see :ref:`pymongo-client-cert` in
+the TLS configuration guide.
+
+Provide a Certificate Key File Password
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. include:: /includes/connect/key-file-password.rst
+
+To learn more about providing a key file password, see :ref:`pymongo-key-file-password` in
+the TLS configuration guide.
+
+Allow Insecure TLS
+~~~~~~~~~~~~~~~~~~
+
+.. include:: /includes/connect/insecure-tls-tabs.rst
+
+To learn more about allowing insecure TLS, see :ref:`pymongo-insecure-tls` in
+the TLS configuration guide.
+
+Disable Certificate Validation
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. include:: /includes/connect/disable-cert-validation-tabs.rst
+
+To learn more about disabling certificate validation, see :ref:`pymongo-insecure-tls` in
+the TLS configuration guide.
+
+Disable Hostname Verification
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. include:: /includes/connect/disable-host-verification-tabs.rst
+
+To learn more about disabling hostname verification, see :ref:`pymongo-insecure-tls` in
+the TLS configuration guide.
+
+Network Compression
+-------------------
+
+Compression Algorithms
+~~~~~~~~~~~~~~~~~~~~~~
+
+.. include:: /includes/connect/compression-tabs.rst
+
+To learn more about specifying compression algorithms, see
+:ref:`pymongo-enable-compression` in the Network Compression guide.
+
+zlib Compression Level
+~~~~~~~~~~~~~~~~~~~~~~
+
+.. tabs::
+
+   .. tab:: MongoClient
+      :tabid: mongoclient
+
+      .. code-block:: python
+
+         client = pymongo.MongoClient("mongodb://<username>:<password>@<hostname>:<port>",
+                                      compressors = "zlib",
+                                      zlibCompressionLevel=<zlib compression level>)
+
+   .. tab:: Connection String
+      :tabid: connectionstring
+
+      .. code-block:: python
+
+         uri = ("mongodb://<username>:<password>@<hostname>:<port>/?"
+                "compressors=zlib"
+                "zlibCompressionLevel=<zlib compression level>")
+         client = pymongo.MongoClient(uri)
+
+To learn more about setting the zlib compression level, see
+:ref:`pymongo-enable-compression` in the Network Compression guide.
+
+Server Selection
+----------------
+
+.. code-block:: python
+
+   client = pymongo.MongoClient("mongodb://<username>:<password>@<hostname>:<port>",
+                                     server_selector=<selector function>)
+
+To learn more about customizing server selection, see
+:ref:`pymongo-server-selection`.
+
+{+stable-api+}
+--------------
+
+.. literalinclude:: /includes/connect/stable_api.py
```

Comment:
Is this is intentionally blank for code generation?

Reply:
It is, but I might as well add it now